test/OpenTelemetry.Contrib.Instrumentation.ElasticsearchClient.Tests/ElasticsearchClientTests.cs

```diff
@@ -180,23 +186,83 @@ public async Task CanCaptureSearchCall()
+            var tags = searchActivity.Tags.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+            Assert.Equal("localhost", searchActivity.GetTagValue(SemanticConventions.AttributeNetPeerName));
+            Assert.Equal(9200, searchActivity.GetTagValue(SemanticConventions.AttributeNetPeerPort));
 
-            Assert.Equal("elasticsearch", searchActivity.GetTagValue(Constants.AttributeDbSystem));
-            Assert.Equal("customer", searchActivity.GetTagValue(Constants.AttributeDbName));
-            var debugInfo = (string)searchActivity.GetTagValue(Constants.AttributeDbStatement);
+            Assert.Equal("elasticsearch", searchActivity.GetTagValue(SemanticConventions.AttributeDbSystem));
+            Assert.Equal("customer", searchActivity.GetTagValue(SemanticConventions.AttributeDbName));
+            var debugInfo = (string)searchActivity.GetTagValue(SemanticConventions.AttributeDbStatement);
             Assert.NotEmpty(debugInfo);
             Assert.Contains("Successful (200) low level call", debugInfo);
 
             Assert.Equal(Status.Unset, searchActivity.GetStatus());
-            Assert.Equal(expectedResource, searchActivity.GetResource());
+
+            // Assert.Equal(expectedResource, searchActivity.GetResource());
+        }
+
+        [Fact]
+        public async Task CanSampleSearchCall()
+        {
+            bool samplerCalled = false;
+
+            var sampler = new TestSampler
+            {
+                SamplingAction =
+                (samplingParameters) =>
+                {
+                    samplerCalled = true;
+                    return new SamplingResult(SamplingDecision.RecordAndSample);
+                },
+            };
+
+            using TestActivityProcessor testActivityProcessor = new TestActivityProcessor();
+
+            bool startCalled = false;
+            bool endCalled = false;
+
+            testActivityProcessor.StartAction =
+                (a) =>
+                {
+                    Assert.True(samplerCalled);
+                    Assert.False(Sdk.SuppressInstrumentation);
+                    Assert.True(a.IsAllDataRequested); // If Proccessor.OnStart is called, activity's IsAllDataRequested is set to true
+                    startCalled = true;
+                };
+
+            testActivityProcessor.EndAction =
+                (a) =>
+                {
+                    Assert.False(Sdk.SuppressInstrumentation);
+                    Assert.True(a.IsAllDataRequested); // If Processor.OnEnd is called, activity's IsAllDataRequested is set to true
+                    endCalled = true;
+                };
+
+            var client = new ElasticClient(new ConnectionSettings(new InMemoryConnection()).DefaultIndex("customer").EnableDebugMode());
+
+            using (Sdk.CreateTracerProviderBuilder()
+                .SetSampler(new AlwaysOffSampler())
+                .AddElasticsearchClientInstrumentation()
```

Comment:
until we get the newer build from main repo (the one with IncrementIfTriggered), can you try 
`.AddElasticsearchClientInstrumentation((opt) => opt.SuppressDownstreamInstrumentation = false)`

Reply:
Once you do this (and fix the sampler -see above comment-), this test should pass...

Reply:
I made both of those changes and it still fails. I had it using the `sampler` instance and then tried the `AlwaysOffSampler` and forgot to put it back. For some reason, it's not calling the `testActivityProcessor.StartAction` callback.

Reply:
Please update the `AddElasticsearchClientInstrumentation` method in `TracerProviderBuilderExtensions.cs` to use `builder.AddLegacyActivity("CallElasticsearch");`.You'll have to update the other unit tests to expect the processor invocations to  equal 5 instead of 4 as the Processor.OnStart should also be called.

Reply:
When is the method name change supposed to be merged into the main repo?